azure/services/agentpools/spec_test.go

```diff
@@ -215,6 +235,39 @@ func TestParameters(t *testing.T) {
 			expected:      sdkFakeAgentPool(),
 			expectedError: nil,
 		},
+		{
+			name: "parameters with an existing agent pool and update needed on scale down mode",
+			spec: fakeAgentPool(),
+			existing: sdkFakeAgentPool(
+				sdkWithScaleDownMode(containerservice.ScaleDownModeDeallocate),
+				sdkWithProvisioningState("Succeeded"),
+			),
+			expected:      sdkFakeAgentPool(),
+			expectedError: nil,
+		},
+		{
+			name: "parameters with an existing agent pool and update needed on spot max price",
+			spec: fakeAgentPool(),
+			existing: sdkFakeAgentPool(
+				sdkWithSpotMaxPrice(123.456),
+				sdkWithProvisioningState("Succeeded"),
+			),
+			expected:      sdkFakeAgentPool(),
+			expectedError: nil,
+		},
+		{
+			name: "parameters with an existing agent pool and update needed on spot max price",
+			spec: fakeAgentPool(
+				withSpotMaxPrice("789.123"),
+			),
+			existing: sdkFakeAgentPool(
+				sdkWithProvisioningState("Succeeded"),
+			),
+			expected: sdkFakeAgentPool(
+				sdkWithSpotMaxPrice(789.123),
+			),
+			expectedError: nil,
+		},
```

Comment:
Can we have some test cases supporting _five decimal places_ ?

Reply:
Sure, done.